synapse/storage/database.py

```diff
@@ -1279,21 +1280,23 @@ async def simple_upsert_many(
             value_names: The value column names
             value_values: A list of each row's value column values.
                 Ignored if value_names is empty.
+            lock: True to lock the table when doing the upsert.
```

Comment:
This should likely say it only applies for emulated?

```suggestion
            lock: True to lock the table when doing the upsert. Unused if the database engine
            	supports native upserts.
```